### PR TITLE
fix: do not try to access undefined variable on client fetch

### PR DESCRIPTION
**Pull request by chisacam**
- resolves https://github.com/lablup/backend.ai-ossca-2023/issues/3
- add check block for `variable is not None`.
- So, that makes correct error response when got ClientResponseError in FetchContextManager.

before

```
✗ [0] cannot access local variable 'raw_resp' where it is not associated with a value
```

after

```
✗ [0] BackendClientError("API endpoint response error.\n➜ ClientResponseError(RequestInfo(url=URL('http://127.0.0.1:6011/session'), method='POST', headers=<CIMultiDictProxy('Host': '127.0.0.1:6011', 'User-Agent': 'Backend.AI Client for Python 23.03.9.dev0', 'X-BackendAI-Domain': 'default', 'X-BackendAI-Version': 'v7.20230615', 'Date': '2023-07-20T08:29:03.767375+00:00', 'Content-Type': 'application/json', 'Authorization': 'BackendAI signMethod=HMAC-SHA256, credential=AKIAIOSFODNN7EXAMPLE:8a8c57ef22208c21eb27c1fe7b68e5e5773de6e21849fd78f2579654c6bf6f4c', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'Content-Length': '543')>, real_url=URL('http://127.0.0.1:6011/session')), (), status=400, message='Expected HTTP/')")
```

### CODE REVIEW

File: docs/dev/development-setup.rst [hunk 1]
```diff
@@ -238,9 +238,8 @@ Open yet another terminal for client and run:
 
 .. code-block:: console
 
-   $ export BACKEND_ENDPOINT=http://127.0.0.1:8081/  # change the port number if customized
-   $ export BACKEND_ACCESS_KEY=AKIAIOSFODNN7EXAMPLE
-   $ export BACKEND_SECRET_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+   $ source ./env-local-admin-api.sh # generated in project root if install-dev.sh worked correctly.
+   $ # source ./env-local-user-api.sh # Choose and use the environment you want.
```
Comment (fregataa) [inline docs/dev/development-setup.rst]: I think that this part is from another branch

**Review by fregataa — CHANGES_REQUESTED** [review]
Thanks, I left a comment!

File: src/ai/backend/client/request.py [hunk 2]
```diff
@@ -573,6 +573,7 @@ def __init__(
     async def __aenter__(self) -> Response:
         max_retries = len(self.session.config.endpoints)
         retry_count = 0
+        raw_resp = None
```
Comment (fregataa) [inline src/ai/backend/client/request.py]: How about add a type hint here?

**Review by fregataa — COMMENTED** [review]
I left one comment.
Plus, please merge the main branch into this branch!

File: src/ai/backend/client/request.py (hunk 2 above)
Comment (chisacam) [reply src/ai/backend/client/request.py]: i added type hint like other part in that file.

**Comment by chisacam** [comment]
sorry, i've rebased instead merge... is it okay?

**Review by kyujin-cho — APPROVED** [review]
Nice catch! Thanks.

**Pull request merged**